contracts/transfer/circuits/src/gadgets/merkle.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details. 
+
+use phoenix_core::note::Note;
+use dusk_plonk::prelude::*;
+use poseidon252::merkle_proof::{merkle_opening_gadget, PoseidonBranch};
+
+pub fn merkle(composer: &mut StandardComposer, branch: PoseidonBranch, note: &Note) {
+
+    let leaf = composer.add_input(note.hash());
+    let root = branch.root;
+
+    merkle_opening_gadget(composer, branch, leaf, root);
+}
```

Comment:
There's no need to have a wrapper for this. Also, you should get the hash directly as a `Variable`.

Reply:
Stills not being solved. Adrress this please.